src/sedcli_main.c

```diff
@@ -277,50 +277,31 @@ int setpw_handle_opts(char *opt, char **arg)
 
 static int get_password(struct sed_key *pwd)
 {
-	int c, flag = 0;
+	size_t dest = SED_MAX_KEY_LEN + 2;
+	uint8_t temp[dest];
 
 	echo_disable();
 
-	if (fgets((char *) pwd->key, SED_MAX_KEY_LEN, stdin) == NULL) {
+	memset(temp, 0, dest);
+
+	if (fgets((char *) temp, dest, stdin) == NULL) {
 		sedcli_printf(LOG_INFO, "Error getting password\n");
 		return -EINVAL;
 	}
-
 	sedcli_printf(LOG_INFO, "\n");
 
-	pwd->len = strnlen((char *) pwd->key, SED_MAX_KEY_LEN);
-
-	/* Handle case when user entered more characters than allowed max size */
-	if (pwd->len == SED_MAX_KEY_LEN - 1 && pwd->key[pwd->len - 1] != '\n') {
-		while ((c = getchar()) != '\n' && c != EOF) {
-			flag = 1;
-			break;
-		}
-
-		if (flag) {
-			memset(pwd->key, 0, SED_MAX_KEY_LEN);
-			pwd->len = 0;
-			sedcli_printf(LOG_ERR, "Password too long\n");
-			echo_enable();
-			return -EINVAL;
-		}
-	}
-
-	/* Remove new line character if needed */
-	if (pwd->key[pwd->len - 1] == '\n') {
-		pwd->key[pwd->len - 1] = 0;
-		pwd->len--;
-	}
-
-	if (pwd->len == 0) {
-		sedcli_printf(LOG_ERR, "Password too short\n");
-		memset(pwd->key, 0, SED_MAX_KEY_LEN);
-		pwd->len = 0;
+	if (temp[dest - 2] != '\n' && temp[dest - 2] != '\0') {
+		sedcli_printf(LOG_INFO, "Password too long..!!!\n");
 		echo_enable();
 		return -EINVAL;
 	}
 
+	pwd->len = strnlen((char *) temp, SED_MAX_KEY_LEN);
+
+	memcpy(pwd->key, temp, SED_MAX_KEY_LEN);
```

Comment:
temp buffer needs to be securely erased after password is copied to the destination buffer

Reply:
Done